libs/dbus_util/inc/public/tfc/dbus/sml_interface.hpp

```diff
@@ -263,6 +284,31 @@ auto get_action_label(const std::string& src_state) -> std::string {
   return action_name.empty() ? "" : fmt::format(R"({} [label = "{}\nentry / {}"])", src_state, src_state, action_name);
 }
 
+inline auto filter_sub_state_machine_id(std::string& state) -> void {
+  std::string_view const sub_state_machine_id{ "boost::sml::back::sm<boost::sml::back::sm_policy<" };
+
+  if (state.starts_with(sub_state_machine_id)) {
+    // remove boost::sml::back::sm<boost::sml::back::sm_policy<
+    state.erase(0, sub_state_machine_id.size());
+    // remove >>
+    state.erase(state.size() - 2, state.size());
+
+    // only take the string up until the first <
+    state.erase(state.find_first_of("<"), state.size());
+
+    // replace :: with _
+    std::string const pattern{ "::" };
+    std::string const replacement{ "_" };
+    size_t pos = 0;
+    while ((pos = state.find(pattern, pos)) != std::string::npos) {
+      state.replace(pos, pattern.length(), replacement);
+      pos += replacement.length();
+    }
+    // place sub_state_machine in front
+    state = fmt::format("sub_sm_{}", state);
+  }
+}
+
```

Comment:
Test if constexpr name of the struct sm

Reply:
added a getter for state name, please see changes.